src/components/DopplerPlus/PlanCalculator.js

```diff
@@ -4,59 +4,80 @@ import { InjectAppServices } from '../../services/pure-di';
 import { Loading } from '../Loading/Loading';
 import { useRouteMatch } from 'react-router-dom';
 import { useIntl } from 'react-intl';
+import queryString from 'query-string';
+import { extractParameter } from '../../utils';
 
-const PlanCalculator = ({ dependencies: { dopplerLegacyClient } }) => {
+const PlanCalculator = ({ location, dependencies: { dopplerLegacyClient } }) => {
+  const safePromoId = extractParameter(location, queryString.parse, 'promoId') || '';
+  const discountId = parseInt(extractParameter(location, queryString.parse, 'discountId')) || 0;
   const { params } = useRouteMatch();
-  const { typePlanId, promoId, discountId } = params;
-  const safePromoId = promoId || '';
-  const safeDiscountId = discountId || 0;
+  const { typePlanId } = params;
   const intl = useIntl();
   const _ = (id, values) => intl.formatMessage({ id: id }, values);
 
   const [state, setState] = useState({ loading: true });
 
-  const [currentPlan, setSelectedPlan] = useReducer(
-    (currentPlan, index) => {
-      return state.planList[index] || currentPlan;
-    },
-    { idPrice: 0, price: 0, amount: 0 },
-  );
+  const actionTypes = {
+    UPDATE_SELECTED_PLAN: 'updateSelectedPlan',
+    UPDATE_SELECTED_DISCOUNT: 'updateSelectedDiscount',
+    INIT: 'init',
+  };
 
-  const [currentDiscount, applyDiscount] = useReducer(
-    (discount, index) => {
-      return state.discountsList[index] || discount;
+  const [planData, dispatchPlanData] = useReducer(
+    (prevPlanData, action) => {
+      switch (action.type) {
+        case actionTypes.UPDATE_SELECTED_PLAN:
+          return { ...prevPlanData, plan: state.planList[action.indexPlan] };
+        case actionTypes.UPDATE_SELECTED_DISCOUNT:
+          return {
+            ...prevPlanData,
+            discount: state.discountsList.find((discount) => {
+              return discount.id === action.idDiscount;
+            }),
+          };
+        case actionTypes.INIT:
+          return {
+            plan: action.plan,
+            discount: action.discount,
+          };
+        default:
+          return prevPlanData;
+      }
     },
-    { id: 0, percent: 0, monthsAmmount: 0, description: '' },
+    { plan: {}, discount: {} },
   );
 
   useEffect(() => {
     const fetchData = async () => {
       setState({ loading: true });
       const responsePlansList = await dopplerLegacyClient.getPlansList(typePlanId);
       if (responsePlansList.success) {
+        dispatchPlanData({
+          type: actionTypes.INIT,
+          plan: responsePlansList.planList[0],
+          discount: discountId
+            ? responsePlansList.discounts.find((discount) => {
+                return discount.id === discountId;
+              })
+            : responsePlansList.discounts[0],
+        });
         setState({
           loading: false,
           planList: responsePlansList.planList,
           discountsList: responsePlansList.discounts,
           success: true,
```

Comment:
No estoy seguro, pero tendrá sentido centralizar todo el estado en un solo reducer en lugar de tener el _useReducer_ y _useState_?

Reply:
I'm not sure neither. One is related with current values and things that may be changed, and the other with information from BE and loading state and so on, but are things that once you have loaded it won't change. It will be a too big reducer if we manage all that states there, but I don't know.

Reply:
Capaz tiene sentido lo que dice, pensalo 😄 los datos de plan mas alla de que los pidas una sola vez estan asociados a la seleccion de plan, podrias probar a ver que pasa, y si no funciona descartar la idea.

Reply:
Lo estuvimos hablando con Andres y ambos creemos que no esta mal tener los dos estados por separado, por un lado los datos fijos y por otro lo que va modificando el usuario.

Ahora voy a aplicar un pequeño cambio para evitar tener tantas dependencias en el useEffect, aunque en realidad me desligo de una sola, pero cuantas menos tengamos mejor.